### PR TITLE
Debounce transporter input

### PR DESCRIPTION
**Pull request by sheckathorne**
## Description
Debounces the transporter input in the manifest form while also reducing the minimum number of characters from 5 to 2 before results begin populating. Debounce will limit the number of requests made to the server when typing-to-search while the reduction in characters will allow the user to display results earlier.

## Issue ticket number and link

https://github.com/USEPA/haztrak/issues/727

## Checklist

- [ x ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ x ] My changes generate no new warnings

### CODE REVIEW

**Comment by dpgraham4401** [comment]
Thanks @sheckathorne!

I know we already discussed this, I'll review this before the end of this week.

File: client/src/hooks/useDebounce/useDebounce.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import '@testing-library/jest-dom';
+import { renderHook, act } from '@testing-library/react';
+import { useDebounce } from './useDebounce';
+import { beforeAll, afterAll, afterEach, describe, expect, it, vi } from 'vitest';
+
+describe('useDebounce hook', () => {
+  beforeAll(() => {
+    vi.useFakeTimers(); // Use fake timers
```
Comment (dpgraham4401) [inline client/src/hooks/useDebounce/useDebounce.spec.tsx:8]: This looks like it was written by AI, which is fine (if it works and isn't doing it in some backwards way), but it doesn't make sense use comments like`// use fake times` to explain a function call called `vi.useFakeTimers()`. same for the rest of the test.

Comment (dpgraham4401) [reply client/src/hooks/useDebounce/useDebounce.spec.tsx:8]: I appreciate the 100% test coverage!
image

**Comment by dpgraham4401** [comment]
I notice that it's still sending a request with an empty epaID when the user moves focus away from the select input, which I believe it was doing before anyways(?)

image

We can remove that call by modifying the skip option in the `useSearchRcraSitesQuery`

```typescript
  const { data } = useSearchRcraSitesQuery(
    {
      siteType: handlerType,
      siteId: debouncedInputValue,
    },
    { skip: skip || debouncedInputValue === '' }
  );
```

@sheckathorne, do you have the time to modify the PR to fix this? I would hate to not clear this technical debt while we're already here. If not I can make these changes.

**Review by dpgraham4401 — CHANGES_REQUESTED** [review]
everything looks good, just some minor edits.

**Comment by sheckathorne** [comment]
Sorry I missed your comments, I have been traveling. I will make these edits today and also admit that I had some help with writing only the tests :)

**Review by dpgraham4401 — APPROVED** [review]
Awesome Possum!
Appreciate the work @sheckathorne

**Pull request merged**